### PR TITLE
Run k6 against local Nuxt on PRs

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Part of https://github.com/WordPress/openverse-infrastructure/issues/1031 by @obulat
Depends on and is blocked by #4908.

## Description

Run k6 against a locally running Nuxt in PRs. This acts as an integration test of the k6 tests themselves, as well as an informal performance test of the local Nuxt application (albeit, without constraints or anything to normalise the results).

I'm not comfortable yet failing this check if any of the k6 requests fail, because I don't know how reliable it will be to run these tests in the GitHub runner with its specific resources constraints. We can evaluate that in this PR and add it now or later if we like.

## Testing Instructions

Check out the new CI step in this PR and see that it makes sense. It should leave a comment with the k6 text summary.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [N/A] I tried running the project locally and verified that there are no visible errors.
- [N/A] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by sarayourfriend** [comment]
Okay, the tests are getting throttled when they make API calls. That makes sense. Because these are only local tests meant to test the frontend (and to ensure the tests themselves work), I think it's appropriate to incorporate talkback here. I'll generate new tapes, and have a commit soon to add that.

**Comment by sarayourfriend** [comment]
Finally! It works! Mostly! Yay! I'll push some last tidy-up changes and then undraft this, though it is still blocked on #4908. @obulat any chance you will have an opportunity to review these monorepo PRs, if not also the infrastructure one?

**Review by obulat — CHANGES_REQUESTED** [review]
I tried running these tests locally, and all of the `fa` locale tests have failed because the `/fa` url shows the "This content has disappeared" error page. Does it run fine for you?

I remember I added the `fa` locale for RTL instead of the `ar` locale to these tests, but don't remember the reason. Will have to look it up.

My main blocker to this PR is the number of lines changed. I know those are due to the added tapes, so I opened a PR to only save the first 200 peaks of the audio response.

**Comment by sarayourfriend** [comment]
Another method of reducing the tape size, is we could minify parts of them, or perhaps the entire tape. It's nice for them to be inspectable _when that's needed_, but really we could probably leave them minified by default... It would certainly make the line counts less surprising when adding new tapes!

**Comment by obulat** [comment]
> Another method of reducing the tape size, is we could minify parts of them, or perhaps the entire tape. It's nice for them to be inspectable _when that's needed_, but really we could probably leave them minified by default... It would certainly make the line counts less surprising when adding new tapes!

I do use the tape contents sometimes, e.g. when trying to find which id is being opened from the search page - to make sure that the tape exists, or to add exactly that tape.

> Yes, but you need to install production translations (which the CI step does using setup-env with locales: "production"). This is a pre-existing issue with these test scripts. If you remember why and we can change it to use the ar so we can use the test locales (or we can import the current fa strings as a new testing locale?) works for me.

Oh, I didn't realize it was using the production locales :)
I _think_ I might have wanted to change a locale that doesn't exist. Maybe we should try to change it back to `ar` and see if anything breaks? I don't think it would break anything, and I don't think we need to test non-existent locale.

File: .github/workflows/ci_cd.yml
Comment (obulat) [inline .github/workflows/ci_cd.yml:671]: Maybe the load testing could be split to a separate action?

**Review by obulat — APPROVED** [review]
Awesome, excited to have the tests running soon 🚀

File: .github/workflows/ci_cd.yml
Comment (sarayourfriend) [reply .github/workflows/ci_cd.yml:671]: Potentially, and it's something I plan to look into when adding the job to run k6 against staging in the next PR after this one. The steps to run k6 on a local environment are so different than running it against a remote environment, I wanted to make sure I didn't abstract too early :slightly_smiling_face:

**Comment by sarayourfriend** [comment]
I'll push a change to use `ar` instead of `fa` and then merge this provided things pass :slightly_smiling_face: 

We can also opt to make locales configurable if we want to run against more or different locales in live environments.

**Pull request merged**
